### PR TITLE
Fix DLNA and HDHR for Traefik Docker compose

### PR DESCRIPTION
**Pull request by PrplHaz4**
Based on the changes proposed by @Artiume 

- Moved Traefik and Jellyfin to a single docker-compose.yml
- Removed some config items
- Added functional host networking so DLNA and LiveTV work
- Used static rules to point to docker host from Traefik network
- Changed docker-compose version to 3.5 (to support network name) - there are other workarounds but this is more copy/paste friendly

### CODE REVIEW

**Comment by PrplHaz4** [comment]
> 
> 
> I think /acme.json is the right mount

I thought so too, but all the Traefik docs show it this way, and it seems to be working for me locally, so I'm going to leave it unless someone reports it being broken.

https://github.com/containous/traefik/blob/v1.7/docs/user-guide/docker-and-lets-encrypt.md

**Review by Artiume — APPROVED** [review]

**Comment by JustAMan** [comment]
@PrplHaz4 please verify that your PR and https://github.com/jellyfin/jellyfin-docs/pull/170 don't clash over DLNA documentation :)

**Comment by PrplHaz4** [comment]
@JustAMan no conflicts that I see - @Artiume and I talked about the changes prior to this PR...

**Review by Artiume — APPROVED** [review]

File: general/administration/reverse-proxy.md [hunk 1]
```diff
@@ -8,13 +8,14 @@ title: Reverse Proxy
...
 
-Some popular options for reverse proxy systems are [Apache](https://httpd.apache.org/), [Haproxy](https://www.haproxy.com/), [Caddy](https://caddyserver.com/), and [Nginx](https://www.nginx.com/).
+Some popular options for reverse proxy systems are [Apache](https://httpd.apache.org/), [Haproxy](https://www.haproxy.com/), [Nginx](https://www.nginx.com/), [Caddy](https://caddyserver.com/) and [Traefik](https://traefik.io/).
 
 **Important:** In order for a reverse proxy to have the maximum benefit, you should have a publically routable IP address and a domain with DNS set up correctly.  These examples assume you want to run Jellyfin under a sub-domain (ie: jellyfin.example.com), but are easily adapted for the root domain if desired. Running Jellyfin in a subpath (example.com/jellyfin/) is supported by the Android and Web clients.
 
 When following this guide, be sure to replace the following variables with your information:
 
   * `DOMAIN_NAME` - Your public domain name to access Jellyfin on (e.g. jellyfin.example.com)
+  * `domain.tld` - The domain name Jellyfin services will run under (e.g. example.com)
```
Comment (oddstr13) [inline general/administration/reverse-proxy.md]: Stick with `example.com/.org/.net` as per [RFC 2606](https://tools.ietf.org/html/rfc2606).

File: general/administration/reverse-proxy.md [hunk 2]
```diff
@@ -239,117 +240,42 @@ DOMAIN_NAME {
 }
 ```
 
-## LetsEncrypt with Certbot
```
Comment (oddstr13) [inline general/administration/reverse-proxy.md]: Is there any good reasons for changing the order of these sections?
As is, it makes the diff much harder to read.

**Review by oddstr13 — CHANGES_REQUESTED** [review]

File: general/administration/reverse-proxy.md (hunk 2 above)
Comment (Artiume) [reply general/administration/reverse-proxy.md]: As the page currently is, I find it harder to read. The Let's Encypt section is compatible with several over RP's so it is sort of confusing to see Apache/Nginx configs again. having it at the bottom of the page helps isolate it. The entire RP page needs to be reworked to break the different sections into their own pages.

Comment (Artiume) [reply general/administration/reverse-proxy.md]: As the page currently is, I find it harder to read. The Let's Encypt section is compatible with several over RP's so it is sort of confusing to see Apache/Nginx configs again. having it at the bottom of the page helps isolate it. The entire RP page needs to be reworked to break the different sections into their own pages.

Comment (Artiume) [reply general/administration/reverse-proxy.md]: As the page currently is, I find it harder to read. The Let's Encypt section is compatible with several over RP's so it is sort of confusing to see Apache/Nginx configs again. having it at the bottom of the page helps isolate it. The entire RP page needs to be reworked to break the different sections into their own pages.

Comment (oddstr13) [reply general/administration/reverse-proxy.md]: I agree that the different reverse proxy solutions need their own pages.
I think that should be done in a separate pr however, as the moving of the section makes reviewing this pr more difficult than it need be.

**Comment by PrplHaz4** [comment]
Sorry @oddstr13 - my intention was to have atomic commits that could be reviewed independently (or thrown out) without making a bunch of PRs. The first commit is the real meat of this issue. 

I'll change to example.com and move reordering to another PR.

Agree they should all be on their own pages - but that will also require someone to dissect lets encrypt into nginx and apache pages (or DRY and put it on its own page).

**Comment by PrplHaz4** [comment]
@oddstr13 I think I'm going back on undoing the ordering - the current order makes no sense and is confusing: 
- Apache
- HAProxy
- Nginx
- Caddy
- Lets Encrypt with Certbot
  - Apache
  - HAProxy
  - Nginx
- Traefik

My change does this (so at least the high level list of reverse proxies is consistent - certbot instructions for each proxy are moved to the bottom):
- Apache
- HAProxy
- Nginx
- Caddy
- Traefik
- Lets Encrypt with Certbot
  - Apache
  - HAProxy
  - Nginx

IMO, this is a reasonable stopgap solution until each RP is moved to its own page (and will hopefully help lots of the issues people are having with Traefik/--net=host.

File: general/administration/reverse-proxy.md (hunk 1 above)
Comment (PrplHaz4) [reply general/administration/reverse-proxy.md]: updated with commit 215dfe9

File: general/administration/reverse-proxy.md (hunk 2 above)
Comment (PrplHaz4) [reply general/administration/reverse-proxy.md]: Reviewing this commit on its own should help (the others are cosmetic): https://github.com/jellyfin/jellyfin-docs/pull/168/commits/a0a9e4887384ab3279fe9894bd8146a9a7b15b08

See my comments below about reordering - I really think it is better in this PR than the original.

**Review by oddstr13 — APPROVED** [review]
I don't know the specifics of docker-compose or Træfik, but looks good to me.

**Pull request merged**